File: lecture0-30/lecture007/find.java
import java.util.Scanner;
public class find{
    public static Scanner scn=new Scanner(System.in);
    public static void main(String[] args){
        
        
        System.out.print("enter row and col value of array : ");
        int row=scn.nextInt();
        int col=scn.nextInt();
        System.out.println("enter array ");
        int [][] arr=new int[row][col];
        input(arr);
        System.out.println("maximum of array is " + maxi(arr));
        System.out.print("enter data to find : ");
        int n;
        n=scn.nextInt();
        System.out.println(find_(arr , n));
        
        }
public static boolean find_(int [][] arr , int data){
    for(int i=0; i<arr.length; i++){
        for(int j=0 ; j<arr.length ; j++){
            if(arr[i][j]==data)
            return true;

        }
}
return false;
}

public static void input(int [][] arr){
    for(int row=0 ; row<arr.length ; row++){
        for(int col=0 ; col< arr.length ; col++){
            arr[row][col]=scn.nextInt();
        }
    }
}
public static int maxi(int [][] arr){
    int max_=arr[0][0];
    for(int row=0 ; row<arr.length ; row++){
        for(int col=0 ; col< arr.length ; col++){
            if(max_<arr[row][col])
            max_=arr[row][col];
        }
    }
return max_;
}
}

File: lecture0-30/lecture007/find.cpp
#include<iostream>
#include<vector>
using namespace std;

void input(vector<vector<int>> &arr ){
    for(int i=0 ; i<arr.size() ; i++){
        for(int j=0 ; j<arr.size() ; j++){
            cin>>arr[i][j];
        }
    }
}

int maxi(vector<vector<int>> &arr){
    int max_=arr[0][0];

    for(int i=0 ; i<arr.size() ; i++){
        for(int j=0 ; j<arr.size() ; j++){
            if(arr[i][j]>max_)
            max_=arr[i][j];
        }
    }    
    return max_;
}

bool find(vector<vector<int>> &arr , int data){
    for(int i=0 ; i<arr.size() ; i++){
        for(int j=0 ; j<arr.size() ; j++){
            if(data==arr[i][j])
            return true ;
        }
    }
    return false;
}


int main(){
    int row , col;
    cout<<"enter rows and coloumn of array : ";
    cin>>row>>col;
    vector<vector<int>> arr(row , vector<int> (col, 0));
    cout<<"enter array "<<endl;
    input(arr);
    cout<<"maximum of array is "<<maxi(arr)<<endl;
    cout<<"enter data to find : ";
    int a;
    cin>>a;
    cout<<find(arr , a);
    //display(arr);
}
